[staging-next] Updates to perlPackages 2023-10

stigtsp opened the pull request:
## Description of changes

Bulk update of perlPackages, based on staging-next

Jobset: https://hydra.nixos.org/jobset/nixpkgs/pr-262304-perl-updates

- Includes patch for `hydra_unstable`: https://github.com/NixOS/hydra/pull/1304

Supersedes #261516

stigtsp committed:
perlPackages.PerlTidy: 20211029 -> 20230912

stigtsp committed:
perlPackages.PodCoverageTrustPod: 0.100005 -> 0.100006

stigtsp committed:
perlPackages.PodElemental: 0.103005 -> 0.103006

stigtsp committed:
perlPackages.PodElementalPerlMunger: 0.200006 -> 0.200007

stigtsp committed:
perlPackages.PodChecker: 1.74 -> 1.75

stigtsp committed:
perlPackages.Plack: 1.0048 -> 1.0050

stigtsp committed:
perlPackages.PodStrip: 1.02 -> 1.100

stigtsp committed:
perlPackages.PodParser: 1.63 -> 1.66

stigtsp committed:
perlPackages.PodSimple: 3.42 -> 3.45

stigtsp committed:
perlPackages.PodSpell: 1.20 -> 1.26

stigtsp committed:
perlPackages.PodEventual: 0.094001 -> 0.094003

stigtsp committed:
perlPackages.PrometheusTiny: 0.008 -> 0.011

stigtsp committed:
perlPackages.PodWeaver: 4.015 -> 4.019

stigtsp committed:
perlPackages.PrometheusTinyShared: 0.024 -> 0.027

stigtsp committed:
perlPackages.ProcBackground: 1.21 -> 1.32

stigtsp committed:
perlPackages.ProcProcessTable: 0.59 -> 0.636

stigtsp committed:
perlPackages.RESTClient: 273 -> 281

stigtsp committed:
perlPackages.RTClientREST: 0.60 -> 0.72

stigtsp committed:
perlPackages.Redis: 1.998 -> 2.000

stigtsp committed:
perlPackages.RegexpGrammars: 1.057 -> 1.058

stigtsp committed:
perlPackages.RoleIdentifiable: 0.007 -> 0.009

stigtsp committed:
perlPackages.RoleTiny: 2.001004 -> 2.002004

stigtsp committed:
perlPackages.RoleHasMessage: 0.006 -> 0.007

stigtsp committed:
perlPackages.RpcXML: 0.80 -> 0.82

stigtsp committed:
perlPackages.SQLSplitStatement: 1.00020 -> 1.00023

stigtsp committed:
perlPackages.ScalarListUtils: 1.55 -> 1.63

stigtsp committed:
perlPackages.SQLTranslator: 1.62 -> 1.63

stigtsp committed:
perlPackages.Sereal: 4.025 -> 5.004

stigtsp committed:
perlPackages.SearchXapian: 1.2.25.4 -> 1.2.25.5

stigtsp committed:
perlPackages.SerealEncoder: 4.025 -> 5.004

stigtsp committed:
perlPackages.SerealDecoder: 4.025 -> 5.004

stigtsp committed:
perlPackages.Specio: 0.46 -> 0.48

stigtsp committed:
perlPackages.SetObject: 1.40 -> 1.42

stigtsp committed:
perlPackages.SpecioLibraryPathTiny: 0.04 -> 0.05

stigtsp committed:
perlPackages.StringFlogger: 1.101245 -> 1.101246

stigtsp committed:
perlPackages.StatisticsDescriptive: 3.0800 -> 3.0801

stigtsp committed:
perlPackages.StringCRC32: 2 -> 2.100

stigtsp committed:
perlPackages.Starman: 0.4015 -> 0.4017

stigtsp committed:
perlPackages.StringErrf: 0.008 -> 0.009

stigtsp committed:
perlPackages.StringRandom: 0.31 -> 0.32

stigtsp committed:
perlPackages.StringFormatter: 0.102084 -> 1.235

stigtsp committed:
perlPackages.StringInterpolate: 0.32 -> 0.33

stigtsp committed:
perlPackages.StringRewritePrefix: 0.008 -> 0.009

stigtsp committed:
perlPackages.StringTruncate: 1.100602 -> 1.100603

stigtsp committed:
perlPackages.SubExporterForMethods: 0.100052 -> 0.100055

stigtsp committed:
perlPackages.SubExporter: 0.987 -> 0.990

stigtsp committed:
perlPackages.StructDumb: 0.12 -> 0.14

stigtsp committed:
perlPackages.StringUtil: 1.31 -> 1.34

stigtsp committed:
perlPackages.SubExporterGlobExporter: 0.005 -> 0.006

stigtsp committed:
perlPackages.SubName: 0.26 -> 0.27

stigtsp committed:
perlPackages.SubInstall: 0.928 -> 0.929

stigtsp committed:
perlPackages.SubHandlesVia: 0.016 -> 0.050000

stigtsp committed:
perlPackages.SubQuote: 2.006006 -> 2.006008

stigtsp committed:
perlPackages.TAPParserSourceHandlerpgTAP: 3.35 -> 3.36

stigtsp committed:
perlPackages.SystemCommand: 1.121 -> 1.122

stigtsp committed:
perlPackages.TclpTk: 1.10 -> 1.11

stigtsp committed:
perlPackages.SyntaxKeywordTry: 0.27 -> 0.29

stigtsp committed:
perlPackages.TemplateTiny: 1.12 -> 1.14

stigtsp committed:
perlPackages.TermTable: 0.015 -> 0.017

stigtsp committed:
perlPackages.TermProgressBar: 2.22 -> 2.23

stigtsp committed:
perlPackages.TermReadLineGnu: 1.36 -> 1.46

stigtsp committed:
perlPackages.TermShell: 0.12 -> 0.13

stigtsp committed:
perlPackages.TermUI: 0.46 -> 0.50

stigtsp committed:
perlPackages.TestClass: 0.50 -> 0.52

stigtsp committed:
perlPackages.TestAbortable: 0.002 -> 0.003

stigtsp committed:
perlPackages.Test2Suite: 0.000155 -> 0.000156

stigtsp committed:
perlPackages.Test2Harness: 1.000152 -> 1.000155

stigtsp committed:
perlPackages.TestDifferences: 0.67 -> 0.70

stigtsp committed:
perlPackages.TestFatal: 0.016 -> 0.017

stigtsp committed:
perlPackages.TestCompile: 2.4.1 -> 3.3.1

stigtsp committed:
perlPackages.TestFileContents: 0.23 -> 0.242

stigtsp committed:
perlPackages.TestDeep: 1.130 -> 1.204

stigtsp committed:
perlPackages.TestManifest: 2.021 -> 2.023

stigtsp committed:
perlPackages.TestMinimumVersion: 0.101082 -> 0.101083

stigtsp committed:
perlPackages.TestLeakTrace: 0.16 -> 0.17

stigtsp committed:
perlPackages.TestInter: 1.09 -> 1.10

stigtsp committed:
perlPackages.TestMockModule: 0.175.0 -> 0.177.0

stigtsp committed:
perlPackages.TestNeeds: 0.002006 -> 0.002010

stigtsp committed:
perlPackages.TestMost: 0.37 -> 0.38

stigtsp committed:
perlPackages.TestOutput: 1.031 -> 1.034

stigtsp committed:
perlPackages.TestNoWarnings: 1.04 -> 1.06

stigtsp committed:
perlPackages.TestPerlTidy: 20200930 -> 20230226

stigtsp committed:
perlPackages.TestRoutine: 0.027 -> 0.031

stigtsp committed:
perlPackages.TestTime: 0.08 -> 0.092

stigtsp committed:
perlPackages.TestScript: 1.26 -> 1.29

stigtsp committed:
perlPackages.TestSynopsis: 0.16 -> 0.17

stigtsp committed:
perlPackages.TestTrailingSpace: 0.0600 -> 0.0601

stigtsp committed:
perlPackages.TestWWWMechanize: 1.54 -> 1.60

stigtsp committed:
perlPackages.TestWarnings: 0.030 -> 0.032

stigtsp committed:
perlPackages.TestVars: 0.014 -> 0.015

stigtsp committed:
perlPackages.TestWarn: 0.36 -> 0.37

stigtsp committed:
perlPackages.TextBalanced: 2.04 -> 2.06

stigtsp committed:
perlPackages.TestXPath: 0.19 -> 0.20

stigtsp committed:
perlPackages.TextBibTeX: 0.88 -> 0.89

stigtsp committed:
perlPackages.TextCSV_XS: 1.44 -> 1.52

stigtsp committed:
perlPackages.TextCSV: 2.00 -> 2.03

stigtsp committed:
perlPackages.TextTable: 1.134 -> 1.135

stigtsp committed:
perlPackages.TextParsewords: 3.30 -> 3.31

stigtsp committed:
perlPackages.TextTemplate: 1.59 -> 1.61

stigtsp committed:
perlPackages.TextMultiMarkdown: 1.000035 -> 1.001

stigtsp committed:
perlPackages.TextLorem: 0.3 -> 0.34

stigtsp committed:
perlPackages.Throwable: 0.200013 -> 1.001

stigtsp committed:
perlPackages.TimeDurationParse: 0.15 -> 0.16

stigtsp committed:
perlPackages.TieCycle: 1.225 -> 1.227

stigtsp committed:
perlPackages.TextTrim: 1.03 -> 1.04

stigtsp committed:
perlPackages.TieFile: 1.05 -> 1.07

stigtsp committed:
perlPackages.TreeDAGNode: 1.31 -> 1.32

stigtsp committed:
perlPackages.TreeSimple: 1.33 -> 1.34

stigtsp committed:
perlPackages.TreeSimpleVisitorFactory: 0.15 -> 0.16

stigtsp committed:
perlPackages.TimeLocal: 1.30 -> 1.35

stigtsp committed:
perlPackages.TypeTinyXS: 0.022 -> 0.025

stigtsp committed:
perlPackages.TypeTiny: 1.012000 -> 2.004000

stigtsp committed:
perlPackages.UNIVERSALrequire: 0.18 -> 0.19

stigtsp committed:
perlPackages.TryTiny: 0.30 -> 0.31

stigtsp committed:
perlPackages.Twiggy: 0.1025 -> 0.1026

stigtsp committed:
perlPackages.UserIdentity: 1.00 -> 1.02

stigtsp committed:
perlPackages.URI: 5.05 -> 5.21

stigtsp committed:
perlPackages.URIdb: 0.20 -> 0.21

stigtsp committed:
perlPackages.UUID4Tiny: 0.002 -> 0.003

stigtsp committed:
perlPackages.URIcpan: 1.007 -> 1.009

stigtsp committed:
perlPackages.W3CLinkChecker: 4.81 -> 5.0.0

stigtsp committed:
perlPackages.Workflow: 1.48 -> 1.62

stigtsp committed:
perlPackages.X11XCB: 0.18 -> 0.20

stigtsp committed:
perlPackages.Version: 0.9928 -> 0.9930

stigtsp committed:
perlPackages.WWWMechanize: 2.03 -> 2.17

stigtsp committed:
perlPackages.XMLLibXSLT: 1.99 -> 2.002001

stigtsp committed:
perlPackages.XSParseSublike: 0.16 -> 0.20

stigtsp committed:
perlPackages.XSParseKeyword: 0.34 -> 0.38

stigtsp committed:
perlPackages.XMLLibXML: 2.0208 -> 2.0209

stigtsp committed:
perlPackages.XMLXPath: 1.44 -> 1.48

stigtsp committed:
perlPackages.YAMLPP: 0.026 -> 0.036

stigtsp committed:
perlPackages.YAMLLibYAML: 0.83 -> 0.88

stigtsp committed:
perlPackages.XXX: 0.35 -> 0.38

stigtsp committed:
perlPackages.YAMLTiny: 1.73 -> 1.74

stigtsp committed:
perlPackages.libintl-perl: 1.32 -> 1.33

stigtsp committed:
perlPackages.enum: 1.11 -> 1.12

stigtsp committed:
perlPackages.ZonemasterCLI: 5.0.1 -> 6.000003

stigtsp committed:
perlPackages.curry: 1.001000 -> 2.000001

stigtsp committed:
perlPackages.ZonemasterLDNS: 3.1.0 -> 3.2.0

stigtsp committed:
perlPackages.podlators: 4.14 -> 5.01

stigtsp committed:
perlPackages.libwwwperl: 6.70 -> 6.72

stigtsp committed:
perlPackages.librelative: 1.000 -> 1.002

stigtsp committed:
perlPackages.libnet: 3.12 -> 3.15

stigtsp committed:
perlPackages.vidir: 0.050 -> 0.052

stigtsp committed:
perlPackages.CompressRawZlib: fix zlib config

stigtsp committed:
perlPackages.libnet: remove patch merged by upstream

stigtsp committed:
perlPackages.ModuleScanDeps: cleanups

stigtsp committed:
perlPackages.FCGI: cleanups

stigtsp committed:
perlPackages.URI: cleanups

stigtsp committed:
perlPackages.PackageStashXS: cleanups

stigtsp committed:
perlPackages.HTTPMessage: cleanups

stigtsp committed:
perlPackages.HTMLForm: cleanups

stigtsp committed:
perlPackages.FFICheckLib: cleanups

stigtsp committed:
perlPackages.SQLSplitStatement: cleanups

stigtsp committed:
perlPackages.AlienBuildPluginDownloadGitLab: init at 0.01

stigtsp committed:
perlPackages.AlienLibxml2: cleanups

stigtsp committed:
perlPackages.CryptPasswdMD5: cleanups

stigtsp committed:
perlPackages.CGISimple: cleanups

stigtsp committed:
perlPackages.StructDumb: cleanups

stigtsp committed:
perlPackages.HTTPDaemon: remove merged patches

stigtsp committed:
perlPackages.EmailSender: cleanups

stigtsp committed:
perlPackages.PPIxUtils: init at 0.003

stigtsp committed:
perlPackages.PerlCritic: cleanups

stigtsp committed:
perlPackages.Moose: cleanups

stigtsp committed:
perlPackages.DBIxClassSchemaLoader: cleanups

stigtsp committed:
perlPackages.DateTime: cleanups

stigtsp committed:
perlPackages.DateTimeHiRes: init at 0.04

stigtsp committed:
perlPackages.DateTimeFormatNatural: cleanups

stigtsp committed:
perlPackages.LWP: cleanups, disable network tests

stigtsp committed:
perlPackages.CatalystPluginSession: cleanups

stigtsp committed:
perlPackages.GD: cleanups

stigtsp committed:
perlPackages.LWPProtocolHttps: update patch for NIX_SSL_CERT_FILE

Update patch that supports NIX_SSL_CERT_FILE env variable so it's
compatible with LWP::Protocol::https 6.11

The new patch does not handle SSL_CERT_FILE or default cert store paths
as this is managed by IO::Socket::SSL.

stigtsp committed:
perlPackages.LWPProtocolHttps: cleanups, enable tests

stigtsp committed:
perlPackages.PrometheusTiny: cleanups

stigtsp committed:
perlPackages.PrometheusTinyShared: cleanups

stigtsp committed:
perlPackages.ServerStarter: disable tests

stigtsp committed:
perlPackages.CryptCurve25519: remove merged patch

stigtsp committed:
perlPackages.MathPari: fix patch url

stigtsp committed:
perlPackages.TestFileContents: cleanup

stigtsp committed:
perlPackages.MetricsAny: cleanups

stigtsp committed:
perlPackages.MathProvablePrime: cleanups

stigtsp committed:
perlPackages.FileTreeCreate: init at 0.0.1

stigtsp committed:
perlPackages.FileFindObject: cleanups

stigtsp committed:
perlPackages.ParseDistname: init at 0.05

stigtsp committed:
perlPackages.PerlPrereqScannerNotQuiteLite: cleanups

stigtsp committed:
perlPackages.XSParseSublike: cleanups

stigtsp committed:
perlPackages.HashOrdered: init at 0.014

stigtsp committed:
perlPackages.UserIdentity: cleanups

stigtsp committed:
perlPackages.PerlPrereqScanner: cleanups

stigtsp committed:
perlPackages.ZonemasterLDNS: cleanups

stigtsp committed:
perlPackages.SubHandlesVia: cleanups

stigtsp committed:
perlPackages.RoleHooks: init at 0.008

stigtsp committed:
perlPackages.MathGMPz: cleanups

stigtsp committed:
perlPackages.DevelCheckOS: cleanups

stigtsp committed:
perlPackages.CSSDOM: fix test

stigtsp committed:
perlPackages.FeatureCompatTry: init at 0.05

stigtsp committed:
perlPackages.ParserMGC: cleanups

stigtsp committed:
perlPackages.SyntaxKeywordTry: cleanups

stigtsp committed:
perlPackages.MailAuthenticationResults: cleanups

stigtsp committed:
perlPackages.ModuleCPANTSAnalyse: cleanups

stigtsp committed:
perlPackages.MojoSQLite: cleanups

stigtsp committed:
perlPackages.FileFindObjectRule: cleanups

stigtsp committed:
perlPackages.TestFutureIOImpl: init at 0.14

stigtsp committed:
perlPackages.FutureIO: cleanups

stigtsp committed:
perlPackages.IOAsync: cleanups

stigtsp committed:
perlPackages.AppMusicChordPro: cleanups

stigtsp committed:
perlPackages.CacheMemcachedFast: cleanups

stigtsp committed:
perlPackages.ConvertColor: cleanups

stigtsp committed:
perlPackages.CryptPerl: cleanups

stigtsp committed:
perlPackages.CryptPKCS10: add test dependency

04_complex_rdn.t depends on Crypt::PK:ECC
https://github.com/openxpki/Crypt-PKCS10/issues/7

stigtsp committed:
perlPackages.DataULID: cleanups

stigtsp committed:
perlPackages.CryptOpenSSLAES: cleanups

stigtsp committed:
perlPackages.TestDiagINC: init at 0.010

stigtsp committed:
perlPackages.CSSMinifierXS: cleanups

stigtsp committed:
perlPackages.GraphicsToolkitColor: init at 1.71

stigtsp committed:
perlPackages.Chart: cleanups

stigtsp committed:
perlPackages.CLIHelpers: cleanups

stigtsp committed:
perlPackages.FileShare: cleanups

stigtsp committed:
perlPackages.JavaScriptMinifierXS: cleanups

stigtsp committed:
perlPackages.NetAsyncHTTP: cleanups

stigtsp committed:
perlPackages.NetAsyncHTTPServer: cleanups

stigtsp committed:
perlPackages.MockMonkeyPatch: init at 1.02

stigtsp committed:
perlPackages.Workflow: cleanups

stigtsp committed:
perlPackages.TextLorem: cleanups

stigtsp committed:
perlPackages.DataCompactReadonly: init at 0.1.0

stigtsp committed:
perlPackages.DevelDeprecationsEnvironmental: init at 1.101

stigtsp committed:
perlPackages.ScalarType: init at 0.3.2

stigtsp committed:
perlPackages.StringBinaryInterpolation: 1.0.0

stigtsp committed:
perlPackages.NumberPhone: cleanups

stigtsp committed:
perlPackages.Curses: fix build

stigtsp committed:
perlPackages.JSONAny: cleanups

stigtsp committed:
perlPackages.IOAsyncSSL: remove merged patch

stigtsp committed:
perlPackages.MooseXTypesURI: cleanups

stigtsp committed:
perlPackages.TestTrailingSpace: cleanups

stigtsp committed:
perlPackages.MHonArc: remove merged patch

stigtsp committed:
perlPackages.TestXPath: cleanups

stigtsp committed:
perlPackages.DBIxConnector: cleanups

stigtsp committed:
hydra_unstable: add patch for Crypt::Passphrase::Argon2

Since the default lengths in Crypt::Passphrase::Argon2 changed from 16
to 32 in in 0.009, some tests that expected the passphrase to be
unchanged started failing.

https://github.com/NixOS/hydra/pull/1304

stigtsp committed:
perlPackages.SoftwareLicense: 0.103014 -> 0.104004

stigtsp committed:
perlPackages.DistZillaPluginTestDistManifest: cleanups

stigtsp committed:
perlPackages.DistZillaPluginTestUnusedVars: cleanups

stigtsp committed:
perlPackages.InlineJava: cleanups

stigtsp committed:
perlPackages.Paranoid: 2.07 -> 2.10

stigtsp committed:
perlPackages.Paranoid: fix patch

stigtsp committed:
perlPackages.X11XCB: remove merged patch

vcunat commented:
Jobset: https://hydra.nixos.org/jobset/nixpkgs/pr-262304-perl-updates#tabs-evaluations

stigtsp committed:
perlPackages.PathTool: remove failing test

Tests in t/taint.t fail due to PATH being set, we assume that this is
fine and remove the test.

  t/taint.t .......... 1/21
  #   Failed test 'cwd() should not explode under taint mode'
  #   at t/taint.t line 31.
  #          got: 'Insecure directory in $ENV{PATH} while running with -T switch at /build/PathTools-3.75/blib/lib/Cwd.pm line 204.
  # '
  #     expected: ''

  #   Failed test 'its return value should be tainted'
  #   at t/taint.t line 32.

  #   Failed test 'fastgetcwd() should not explode under taint mode'
  #   at t/taint.t line 31.
  #          got: 'Insecure directory in $ENV{PATH} while running with -T switch at /build/PathTools-3.75/blib/lib/Cwd.pm line 204.
  # '
  #     expected: ''

  #   Failed test 'its return value should be tainted'
  #   at t/taint.t line 32.
  # Looks like you failed 4 tests of 21.

stigtsp committed:
perlPackages.WWWCurl: update patch

stigtsp committed:
Merge branch 'staging-next' into perl-updates

stigtsp commented:
Just merged the latest `staging-next` changes into `perl-updates`, sorry for the pings.

stigtsp committed:
Merge branch 'staging-next' into perl-updates

dasJ reviewed (approved):
Scrolled through the diff. Looks reasonable. Didn't test anything but the Hydra jobset should have caught everything.

stigtsp committed:
perlPackages.Test2Harness: fix tests

Set AUTOMATED_TESTING=1 in the checkPhase as we override it for
Test2Harness to allow it to test itself.

This env var is usually set by default by buildPerlPackage.
